src/main/java/frc/robot/superstructure/Superstructure.java

```diff
@@ -127,71 +173,119 @@ public Command speaker() {
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.SHOOTER_TO_ELEVATOR),
+					funnelStateHandler.setState(FunnelState.SHOOTER_TO_ELEVATOR)
+				),//.until(isNoteInShooter)
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.STOP),
+					funnelStateHandler.setState(FunnelState.STOP)
+				)
+			),
 			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
-			flywheelStateHandler.setState(FlywheelState.DEFAULT),
 			elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_TO_ELEVATOR),
-			funnelStateHandler.setState(FunnelState.SHOOTER_TO_ELEVATOR),
-			intakeStatesHandler.setState(IntakeStates.SHOOTER_TO_ELEVATOR),
-			pivotStateHandler.setState(PivotState.UP),
-			elevatorStatesHandler.setState(ElevatorStates.IDLE)
+			flywheelStateHandler.setState(FlywheelState.DEFAULT),
+			elevatorStatesHandler.setState(ElevatorStates.IDLE),
+			pivotStateHandler.setState(PivotState.UP)
 		);
 	}
 
 	public Command transferElevatorShooter() {
 		return new ParallelCommandGroup(
+			setCurrentStateName(RobotState.TRANSFER_ELEVATOR_SHOOTER),
+			new SequentialCommandGroup(
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.STOP),
+					funnelStateHandler.setState(FunnelState.STOP)
+				),//.until(isReadyToTransfer)
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.ELEVATOR_TO_SHOOTER),
+					funnelStateHandler.setState(FunnelState.NOTE_TO_SHOOTER)
+				),//.until(isNoteInShooter)
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.STOP),
+					funnelStateHandler.setState(FunnelState.STOP)
+				)
+			),
 			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
+			elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_TO_ELEVATOR),
 			flywheelStateHandler.setState(FlywheelState.DEFAULT),
-			elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_FROM_ELEVATOR),
-			funnelStateHandler.setState(FunnelState.NOTE_TO_SHOOTER),
-			intakeStatesHandler.setState(IntakeStates.NOTE_TO_SHOOTER),
-			pivotStateHandler.setState(PivotState.UP),
-			elevatorStatesHandler.setState(ElevatorStates.IDLE)
+			elevatorStatesHandler.setState(ElevatorStates.IDLE),
+			pivotStateHandler.setState(PivotState.UP)
 		);
 	}
 
 	public Command intakeOuttake() {
 		return new ParallelCommandGroup(
+			setCurrentStateName(RobotState.INTAKE_OUTTAKE),
 			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
-			flywheelStateHandler.setState(FlywheelState.DEFAULT),
 			elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_FROM_ELEVATOR),
+			flywheelStateHandler.setState(FlywheelState.DEFAULT),
 			funnelStateHandler.setState(FunnelState.INTAKE_OUTTAKE),
 			intakeStatesHandler.setState(IntakeStates.OUTTAKE),
 			pivotStateHandler.setState(PivotState.ON_FLOOR),
 			elevatorStatesHandler.setState(ElevatorStates.IDLE)
-		);
+		).until(() -> !isNoteInIntake());
```

Comment:
if the note starts in the funnel, shooter or elevator the command will end immediately